pkg/apis/vpc/v1alpha1/subnet_types.go

```diff
@@ -36,7 +40,6 @@ type SubnetSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	IPAddresses []string `json:"ipAddresses,omitempty"`
 	// DHCPConfig DHCP configuration.
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	DHCPConfig DHCPConfig `json:"DHCPConfig,omitempty"`
```

Comment:
I thought NSX want to rename this field. No?

Reply:
The nsx api changes haven't been merged, but the current patch shows that they will introduce a new field called subnet_dhcp_config for the new changes while the old one named dhcp_config. Thus I keep the same field name to make the change in CRD definition small. Would you think we still need to rename the field if NSX side does not use a name with a big change? Ref: https://gitreview-vcfnet.lvn.broadcom.net/c/nsx/+/657239/9/mp/policy/policy-api/src/main/resources/api_spec/root/nsx-policy/policy-vpc-subnet.yml#292

Reply:
Yes, please always keep most consistence with NSX API whenever possible. Please watch NSX API changes.